### ISSUE TITLE
[Feature]: Adding support for synchronization with Piped.

### ISSUE THREAD

**Issue by MattSolo451** [fast4x/RiMusic#278]
### Description
It would be nice if RiMusic gave the option to sync playlists with Piped as it does with Harmony Music. This would give a little more choice and convenience. I'm sure more than one user will agree with me

### Suggested Solution
Adding logging functionality to different Piped instances and piped link support.

**Comment by fast4x** [comment]
Hi, can you explain how you expect Piped integration to work?

**Comment by MattSolo451** [comment]
In terms of the implementation itself, unfortunately I can't say much, but in my opinion it could look twofold.

1) The choice of APIs at the start or in the settings of the application because probably simultaneous use of 2 at the same time is not possible.

2) A separate installer for a specific API. Which could also have the advantage that it would not increase the size of the application and at the same time also give a choice.

**Comment by n-ce** [comment]
AFAIK Harmony Music only uses the piped API for synchronizing saved playlists.
https://github.com/anandnet/Harmony-Music/wiki/Piped-linking

**Comment by MattSolo451** [comment]
I didn't even know until the end. Maybe I'll change the feature request to Support for Synchronization with Piped.

**Comment by NeeRaj-2401** [comment]
in the world of open source projects, we are loosing simplicity in an app :(
ViMusic was example for simple and elegant music streaming app, rest all like harmony, simp etc they are well flooded with unnecesssary thing IMO

PS: relevant to the issue

**Comment by fast4x** [comment]
> in the world of open source projects, we are loosing simplicity in an app :(
> ViMusic was example for simple and elegant music streaming app, rest all like harmony, simp etc they are well flooded with unnecesssary thing IMO
> 
> PS: relevant to the issue
> 

Hello, I respect your point of view, give me some examples on useless things. You can use RiMusic as if it were ViMusic from the settings. ViMusic remains a landmark for me.

**Comment by NeeRaj-2401** [comment]
Apologies if you got me wrong, i wasn't talking about RiMusic, as of now it having decent UI/UX, after viMusic i finally found a alternative to switch to.

i was talking about [Harmony](https://github.com/anandnet/Harmony-Music)'s piped integration thing, then [SimpMusic](https://github.com/maxrave-dev/SimpMusic)'s spotify ui approach with google account login , 

then coming to [LibreTube](https://github.com/libre-tube/LibreTube), it have tons of customization options almost every ui/behavior, & refering to [this](https://github.com/libre-tube/LibreTube/pull/5392#issuecomment-1869630566) comment, i think people ask for way more things than a app should have.

in conclusion we have piped, invidious, sponsorblock, deArrow and so on, people keep asking for many things which leads to increase unecessary complexity in an app IMO, hence i marked ViMusic as an example for its simplicity and serving the purpose nothing more nothing less.

btw Thanks for continuing the project :)  ,

**Comment by fast4x** [comment]
Sorry, I didn’t understand. I think you’re right, in many ViMusic forks and not only, they often ask themselves the same things and unfortunately become quite useless. RiMusic must not lose that simplicity inherited from ViMusic while looking to the future and new implementations.

Thanks for the appreciation

**Comment by MattSolo451** [comment]
> in the world of open source projects, we are loosing simplicity in an app :( ViMusic was example for simple and elegant music streaming app, rest all like harmony, simp etc they are well flooded with unnecesssary thing IMO
> 
> PS: relevant to the issue

It depends on what functions. The function of synchronization even simple between applications on different devices is very useful and does not have to involve an unknown increase in the complexity of the application. But you're right I'll delete this whole post since it bothers you simple man.

**Comment by fast4x** [comment]
Do not delete it, please keep it open. Comments were not referring to Piped.

**Comment by MattSolo451** [comment]
> Do not delete it, please keep it open. Comments were not referring to Piped.

This is what I don't understand. The post is about adding synchronization that would help people using, for example, Libretube to have their own "YouTube Music" meanwhile you write that people are asking for too many features for a particular app. How was that not supposed to be relevant to the discussion?

**Comment by fast4x** [comment]
I understand, but spoke in general, do not take it, everyone has their own ideas.

**Comment by MattSolo451** [comment]
> I understand, but spoke in general, do not take it, everyone has their own ideas.

It might be better to talk about the topic and not discuss other topics. I suggest setting up a group on Matrix so that everyone can chat about their views on general topics about the application.

**Comment by NeeRaj-2401** [comment]
> It might be better to talk about the topic and not discuss other topics

In my very first comment in this thread, i mentioned that this is relevant to this issue, because i simply disagree with such things, 

Despite of that if dev fast4x, would still decide to implement it  **and other such requests** in future then its fine.

I was just trying to make realise that its better to not loose the simplicity rather than relying on people's demand (which is not going to be fulfilled in any time soon),

**Comment by fast4x** [comment]
I like to listen to people, everyone has their own ideas and constructive confrontation is always welcome. The development of the required features is organized by priorities that are managed over time compatible with the general idea of the project.

**Comment by MattSolo451** [comment]
> > It might be better to talk about the topic and not discuss other topics
> 
> In my very first comment in this thread, i mentioned that this is relevant to this issue, because i simply disagree with such things,
> 
> Despite of that if dev fast4x, would still decide to implement it **and other such requests** in future then its fine.
> 
> I was just trying to make realise that its better to not loose the simplicity rather than relying on people's demand (which is not going to be fulfilled in any time soon),

It is not necessary to implement everything at once. You can choose specific things that are useful. For example, I can say that the creation of covers with video in RiMusic is hopeless compared to Harmony Music as well in the native player as in Android auto. In my opinion, it would have to be improved a bit or add the option to decide and customize the appearance of the cover yourself. The other person less aesthete will not mind it. That is to say, it is an unnecessary feature and does not need to be implemented, or rather useful and can be added at a small cost of resources? In my opinion, it's an option just as in the market most people will choose boring passionless cars from the Volkswagen group and the other may, for example, choose Italian fantasy, excitement and style of Alfa Romeo 🤷. Better to have choices and options than to be ranked.

**Comment by Warden20** [comment]
This feature would be useful for people who want a way to synchronize different devices the only thing missing from this application

**Comment by Seryoga1984** [comment]
> ### Description
> It would be nice if RiMusic gave the option to sync playlists with Piped as it does with Harmony Music. This would give a little more choice and convenience. I'm sure more than one user will agree with me
> 
> ### Suggested Solution
> Adding logging functionality to different Piped instances and piped link support.

This function would be really great. A great idea.

**Comment by NikunjKhangwal** [comment]
> Apologies if you got me wrong, i wasn't talking about RiMusic, as of now it is having decent UI/UX, after viMusic i finally found a alternative to switch to.
> 
> i was talking about [Harmony](https://github.com/anandnet/Harmony-Music)'s piped integration thing, then [SimpMusic](https://github.com/maxrave-dev/SimpMusic)'s spotify ui approach with google account login ,
> 
> then coming to [LibreTube](https://github.com/libre-tube/LibreTube), it have tons of customization options almost every ui/behavior, & refering to [this](https://github.com/libre-tube/LibreTube/pull/5392#issuecomment-1869630566) comment, i think people ask for way more things than a app should have.
> 
> in conclusion we have piped, invidious, sponsorblock, deArrow and so on, people keep asking for many things which leads to increase unecessary complexity in an app IMO, hence i marked ViMusic as an example for its simplicity and serving the purpose nothing more nothing less.
> 
> btw Thanks for continuing the project :) ,

You can't deny a feature you don't use. I get where you're coming from but again it's a matter of personal preference. I agree things like minor ui changes create a lot of overhead for developers but also in this issue, something like login is particularly useful for people who want their data synchronized. It's ok if you prefer simplicity but also you gotta realise a lot of people (specially power users) like having versatility and control with multiple features. In the end it's developer choice for which approach they want to take.
On a side note I'm curious to know what you find problematic with SimpMusic's Spotify like ui? For me i think of it as a great option for people who're switching from Spotify even though I'm not going to use it myself and also Spotify UI is just loved by most people in general.

**Comment by fast4x** [comment]
I have the utmost respect for all the requests of all people, what you need to understand is that in the management of a project there are different priorities. Piped is not a priority for this project. Simple!

Login YouTube or other login types are not planned in this project.

**Comment by NikunjKhangwal** [comment]
> I have the most respect for all the requests of all people, what you need to understand is that in the management of a project there are different priorities. Piped is not a priority for this project. Simple!
> 
> Login YouTube or other login types are not planned in this project.

That's a Fair point
The main point was people having preferences. I mentioned it's on developer to choose what they want (and what they priotize too).

**Comment by Cszyn** [comment]
I'd like to bring this up again. I have really wanted to abandon YouTube music's app, but the total lack of synchronization is an absolute dealbreaker I have been hopping around different plays the past week but none offer as smooth of an experience as RI. (Innertune has broken tabs, Simpmusic constantly bricks itself, Harmony's UI is ass.) (I'm using innertune rn.) Likewise, I understand the desire to keep bloat to a minimum and I totally respect that ultimate I'm not the developer, and they're free to do what they want with the app. As a side note, I have seen argument that sync is bloating the app, but I just flat out disagree

**Comment by Cszyn** [comment]
And no i don't want google login/trackers. But anything else

**Comment by fast4x** [comment]
Thanks for your appreciation.

**Comment by fast4x** [comment]
Started development of piped account and synchronization 🎉

**Comment by NikunjKhangwal** [comment]
Finally I can ditch Spotify completely and keep my data synced 🥳

**Comment by Cszyn** [comment]
> Started development of piped account and synchronization 🎉

https://media.tenor.com/VFn1sjf7knsAAAAM/wooo-yeah-baby.gif

**Comment by fast4x** [comment]
Implemented piped playlists syncronization, available from version 0.6.39

**Comment by Cszyn** [comment]
We just ended harmony with this one 🗣️🔥🔥🔥

**Comment by siggi1984** [comment]
> Implemented piped playlists syncronization, available from version 0.6.39

I was actually expecting a piped synchronization like Harmony Music.

**Comment by fast4x** [comment]
Login in piped account in miscellaneous settings

**Comment by siggi1984** [comment]
> Login in piped account in miscellaneous settings

I don't mean just read access. I thought we would also have write access to create playlists, edit them, etc. ...

**Comment by fast4x** [comment]
Actually it's possible because teat for writing it's failed, so I'm developing and testing if possible.

**Comment by siggi1984** [comment]
> Actually isn't possible because teat for writing it's failed, so I'm developing and testing if possible.

It works with Harmony-Music, maybe we can learn something from it???

**Comment by fast4x** [comment]
Yes

**Comment by jaker-dotcom** [comment]
Hello there,
I am very late to the discussion, as I just discovered the app. 
Instead of creating a new issue (because unsure whether there even is one) I wanted to ask here how to set it up.

I interpret your last comment @fast4x so that there is write access.
I set up an account in the settings but yet my playlists don't show on piped in the web and they are filtered out when filtering for "piped". 
Was all very recent (~10min) so maybe it'll just take some time. I restarted the app already.
But since I did not find any documentation on the sync feature I would kindly ask for such so to know what I can expect.

Something short would suffice, like this but with questions answered:

- Why? : Sync playlists to listen to same playlists on multiple devices / apps.
- What does it do? : sync (existing?) playslists (bidirectionally?)
- Where are the settings? : In Miscellaneous tab after enabling piped synchronization there, select an instance and enter your account credentials for that instance

**Comment by fast4x** [comment]
Hi,
after connecting your piped account, you can create a playlist and it will be synced directly. Existing playlists are not synchronized because they are not of type piped.

In the future there will be a feature that allows you to convert a playlist not piped into a piped.

Currently if you want to sync a playlist you have to create new and use the Add to playlist feature from the source playlist, so that you can copy songs from the normal playlist to the piped one. This will sync it to piped, after that you can delete the playlist

**Comment by jaker-dotcom** [comment]
Tried. Newly created playlists don't show up once I hit the filter "piped" either. And I don't see them in the web neither.

Should I maybe try with another instance?

**Comment by fast4x** [comment]
Yes try, you are sure that piped account is connected without problems?

**Comment by fast4x** [comment]
> Will try. How can I be sure?

1. Go to https://piped.video 
2. create a new playlist
3. Go to RiMusic playlists
4.if show new playlist, connection it's ok
